pubsub/cloud-client/quickstart/sub_test.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import mock
+import os
+import pytest
+import time
+
+from google.cloud import pubsub_v1
+
+import sub
+
+
+PROJECT = os.environ['GCLOUD_PROJECT']
+TOPIC = 'quickstart-sub-test-topic'
+SUBSCRIPTION = 'quickstart-sub-test-topic-sub'
+
+
+@pytest.fixture(scope='module')
+def publisher_client():
+    yield pubsub_v1.PublisherClient()
+
+
+@pytest.fixture(scope='module')
+def topic(publisher_client):
+    topic_path = publisher_client.topic_path(PROJECT, TOPIC)
+
+    try:
```

Comment:
Ditto. Seems to me that we should delete after `yield`. Plus choose a random topic name in case multiple tests are running in parallel (such as by CI against multiple versions of Python).

Reply:
Same as above. I added some cleanup code in test. So far I haven't run into problems with CI against multiple version of Python.

Reply:
Updated. I used a pytest fixture `to_delete` to collect resources to be deleted this time.